eservice/pdo/eservice/scripts/EServiceEnclaveInfoCLI.py

```diff
@@ -103,37 +103,14 @@ def LocalMain(config, spid, save_path) :
 ## XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```

Comment:
Could you please move lines 93-94 right after line 78?

Reason:
if there is an exception (like the one currently happening at line 88), the enclave is still up and some workers exit after the logger is unavailable, which makes them generate exceptions.

Reply:
I don't know what this means

Reply:
if i'm understanding you correctly... most of your code for getting IAS certs should be wrapped in a try/catch to avoid this problem. for example, if the directory isn't readable or the json isn't parsable you're still going to crash.

Reply:
Surely the code must be made more robust.
This involves making sure that the enclave workers are shut down before the python logger is not available anymore -- I guess this can only happen when we have unhandled exceptions.

Reply:
Its more complicated than that. The module for the enclaves is set up on shutdown to remove the back references to the log functions. Those need to be handled correctly when the logging shuts down first. It might be best to replace/augment the try/except/finally with an atexit registration. Although... even that might be problematic since controlling the order in which the atexit functions execute is a bit tricky. For now... just use good practice... any time the enclave is started it should operate in a try/except/finally that shuts it down cleanly. I've added that code here for now.